numba/runtime/_nrt_python.c

```diff
@@ -0,0 +1,539 @@
+#include "_pymodule.h"
+#include "nrt.h"
+
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#include <numpy/ndarrayobject.h>
+#include <numpy/arrayscalars.h>
+
+#include "_arraystruct.h"
+
+/* For Numpy 1.6 */
+#ifndef NPY_ARRAY_BEHAVED
+    #define NPY_ARRAY_BEHAVED NPY_BEHAVED
+#endif
+
+
+static
+PyObject*
+memsys_shutdown(PyObject *self, PyObject *args) {
+    if (!PyArg_ParseTuple(args, "")) {
+        return NULL;
+    }
+    NRT_MemSys_shutdown();
+    Py_RETURN_NONE;
+}
+
+static
+PyObject*
+memsys_set_atomic_inc_dec(PyObject *self, PyObject *args) {
+    unsigned PY_LONG_LONG addr_inc, addr_dec;
```

Comment:
I'm not sure PY_LONG_LONG is always enough for a pointer, but it probably is on common architectures... Otherwise, the right way would be PyLong_AsVoidPtr().